specs/sharding/beacon-chain.md

```diff
@@ -101,390 +68,158 @@ The following values are (non-configurable) constants used throughout the specif
+    # The sizes of the blocks encoded in the commitments (last intermediate and all beacon blocks since)
+    included_block_sizes: List[uint64, MAX_BEACON_BLOCKS_BETWEEN_INTERMEDIATE_BLOCKS + 1]
+    
+    # Number of commitments that are for sharded data (no blocks)
+    included_sharded_data_commitments: uint64
 
-```python
-class ShardBlob(Container):
-    slot: Slot
-    shard: Shard
-    # Builder of the data, pays data-fee to proposer
-    builder_index: BuilderIndex
-    # Proposer of the shard-blob
-    proposer_index: ValidatorIndex
-    # Blob contents
-    body: ShardBlobBody
+    # Random evaluation of beacon blocks + execution payload (this helps with quick verification)
+    block_verification_kzg_proof: KZGCommitment
 ```
 
-### `ShardBlobHeader`
-
-Header version of `ShardBlob`.
+#### `SignedShardSample`
 
 ```python
-class ShardBlobHeader(Container):
+class SignedShardSample(Container):
     slot: Slot
-    shard: Shard
-    # Builder of the data, pays data-fee to proposer
-    builder_index: BuilderIndex
-    # Proposer of the shard-blob
-    proposer_index: ValidatorIndex
-    # Blob contents, without the full data
-    body_summary: ShardBlobBodySummary
-```
-
-### `SignedShardBlob`
-
-Full blob data, signed by the shard builder (ensuring fee payment) and shard proposer (ensuring a single proposal).
-
-```python
-class SignedShardBlob(Container):
-    message: ShardBlob
+    row: uint64
+    column: uint64
+    data: Vector[BLSFieldElement, FIELD_ELEMENTS_PER_SAMPLE]
+    proof: KZGCommitment
+    builder: ValidatorIndex
     signature: BLSSignature
 ```
 
-### `SignedShardBlobHeader`
-
-Header of the blob, the signature is equally applicable to `SignedShardBlob`.
-Shard proposers can accept `SignedShardBlobHeader` as a data-transaction by co-signing the header.
-
-```python
-class SignedShardBlobHeader(Container):
-    message: ShardBlobHeader
-    # Signature by builder.
-    # Once accepted by proposer, the signatures is the aggregate of both.
-    signature: BLSSignature
-```
+### Extended Containers
 
-### `PendingShardHeader`
+#### `BeaconState`
 
 ```python
-class PendingShardHeader(Container):
-    # The commitment that is attested
-    attested: AttestedDataCommitment
-    # Who voted for the header
-    votes: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
-    # Sum of effective balances of votes
-    weight: Gwei
-    # When the header was last updated, as reference for weight accuracy
-    update_slot: Slot
-```
-
-### `ShardBlobReference`
-
-Reference version of `ShardBlobHeader`, substituting the body for just a hash-tree-root.
-
-```python
-class ShardBlobReference(Container):
-    slot: Slot
-    shard: Shard
-    # Builder of the data
-    builder_index: BuilderIndex
-    # Proposer of the shard-blob
-    proposer_index: ValidatorIndex
-    # Blob hash-tree-root for slashing reference
-    body_root: Root
+class BeaconState(bellatrix.BeaconState):
```

Comment:
Simply inheritance seems incorrect since `SLOTS_PER_EPOCH` changed and then the field `BeaconState.eth1_data_votes: List[Eth1Data, EPOCHS_PER_ETH1_VOTING_PERIOD * SLOTS_PER_EPOCH]` also need to be extended.